src/markdown/parser.cr

```diff
@@ -452,6 +463,36 @@ class Markdown::Parser
     String.new(Slice.new(str + bracket_idx + 2, paren_idx - 1))
   end
 
+  # Checks if the following sequence of characters is a checkbox.
+  # A valid checkbox is made up of 3 characters and looks like
+  # this `[ ]` or this `[x]` with the latter being a checked one.
+  def check_checkbox(str, pos, bytesize)
```

Comment:
Good point. I figured someone with more knowledge of the markdown parser would be able to give me some helpful feedback since before yesterday I didn't know anything about it.